File: common/src/main/java/com/ssblur/scriptor/word/subject/DebugSubject.java
package com.ssblur.scriptor.word.subject;

import com.ssblur.scriptor.api.word.Subject;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.word.Spell;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public class DebugSubject extends Subject {

  @Override
  public Cost cost() { return new Cost(1, COSTTYPE.ADDITIVE); }

  @Override
  public CompletableFuture<List<Targetable>> getTargets(Targetable caster, Spell spell) {
    var result = new CompletableFuture<List<Targetable>>();
    result.complete(List.of());
    return result;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/subject/DebugSubject.kt
package com.ssblur.scriptor.word.subject

import com.ssblur.scriptor.api.word.Subject
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.Spell
import java.util.concurrent.CompletableFuture

class DebugSubject : Subject() {
    override fun cost(): Cost {
        return Cost(1.0, COSTTYPE.ADDITIVE)
    }

    override fun getTargets(caster: Targetable, spell: Spell): CompletableFuture<List<Targetable>> {
        val result = CompletableFuture<List<Targetable>>()
        result.complete(listOf())
        return result
    }
}
